### DISCUSSION TITLE
🛣️ Roadmap Update - Jan 2022

### DISCUSSION THREAD

**Discussion by lissy93** [Lissy93/dashy#405] [Project Updates]
Hey everyone, I hope you all had a great Christmas/ Winter break! Just a quick update, below are the next 10 upcoming features I will be working on over the next few weeks/ months.

- [ ] **Action Buttons**
  _Similar to items, but rather than opening an app, they will execute commands, via web hooks. This was suggested in [#383](https://github.com/Lissy93/dashy/issues/383)_
- [x] **Sub-Items**
  _The option for folder-items that will contain a series of sub-items or actions_
- [x] **Improved Config Loading**
  _Currently the app is compiled using your config file, which means it rebuilds after config changes. This will be updated, so that config can be read dynamically, while also maintaining performance_
- [x] **Multi-Page Support**
  _This has been a long time coming, but multi-page support will allow you to have multiple dashboards under one instance of Dashy_
- [ ] **Dynamic Widgets**
  _As well as developing a series of new widgets for integrating with self-hosted services, I am also going to make it easier for users to display data from any API-enabled service_
- [ ] **Updates to Config Editor**
  _Implement drag-and-drop for sections and items within the config editor, and functionality for adding + editing widgets through the UI_
- [ ] **Improved section layout**
  Arranging sections of varying sizes can be quite cumbersome, so a masonry-based layout will be built to automatically position sections based on available space_
- [ ] **Lite Version**
  _The app is served up using a Node server. Node.js is quite heavy, so a lite version (using NGINX) will be aimed at lower-spec devices, or users who don't need any of the server features (status checking and UI config editor)_
- [ ] **Docker Updates**
  _Automated builds from a single multi-arch image. And improved privilege security as outlined in [#383](https://github.com/Lissy93/dashy/issues/340) (help needed)_
- [ ] **Dashy API**
  _Develop a simple RESTful JSON API so third-party applications can fetch section and item data. This will make it possible to interact with Dashy data in other form-factors, such as browser extension, mobile app, etc_

Thank you to you all for using Dashy, and especially those who've contributed to the project in someway, it means the world to me :)

Any questions, suggestions, comments, feel free to drop them below,
And hope you all have a very happy new year :)

**Comment by EVOTk** [comment]
Hello Lissy93 ,
Do you plan to develop Widgets for apps like rutorrent/radarr/sonarr/overserr/tautulli?

Sonarr : https://github.com/Sonarr/Sonarr/wiki/API
Radarr : https://radarr.video/docs/api/
Tautulli : https://github.com/Tautulli/Tautulli/wiki/Tautulli-API-Reference
Overseerr : https://api-docs.overseerr.dev/ 

For Overseerr, although I'm not a developer, the API seems "easy". ( it is also possible, through overseerr, to retrieve information on Plex, Tautulli, Sonarr, Radarr, .. directly. ) For example the command, 
```
curl -X 'GET' \
  'https://url_overseerr/api/v1/request/count' \
  -H 'accept: application/json' \
  -H 'X-Api-Key: MY_API_KEY'
```
Returns interesting information:
```{"total":55,"movie":42,"tv":13,"pending":3,"approved":52,"declined":0,"processing":27,"available":25}```

For example, the number of total requests (55), the number awaiting validation (3), or the number of requests in progress (27).

It is also possible to follow the problems reported on overseerr, for example with the command:
```
curl -X 'GET' \                                                                                                                     root@debian-web [17:49]
  'https://url_overseerr/api/v1/issue?take=20&skip=0&sort=added&filter=open&requestedBy=1' \
  -H 'accept: application/json' \
-H 'X-Api-Key: MY_API_KEY'
```
Returns interesting information:
```
{"pageInfo":{"pages":1,"pageSize":20,"results":1,"page":1},"results":[{"id":2,"issueType":1,"status":1,"problemSeason":0,"problemEpisode":0,"createdAt":"2022-02-14T16:50:11.000Z","updatedAt":"2022-02-14T16:50:11.000Z","createdBy":{"permissions":2,"id":1,"email":"XXXXXXXXXXXXXX@XX.COM","plexUsername":"XXXXXX","username":null,"recoveryLinkExpirationDate":null,"userType":1,"avatar":"https://plex.tv/users/XXXXXXXXXXXXXX/avatar?c=XXXXXXXXXXXXXX","movieQuotaLimit":null,"movieQuotaDays":null,"tvQuotaLimit":null,"tvQuotaDays":null,"createdAt":"2022-02-05T21:33:34.000Z","updatedAt":"2022-02-13T22:54:06.000Z","requestCount":41,"displayName":"EVOTk"},"media":{"downloadStatus":[],"downloadStatus4k":[],"id":3491,"mediaType":"movie","tmdbId":476669,"tvdbId":null,"imdbId":null,"status":5,"status4k":1,"createdAt":"2022-02-09T17:50:40.000Z","updatedAt":"2022-02-10T04:05:00.000Z","lastSeasonChange":"2022-02-09T17:50:40.000Z","mediaAddedAt":"2022-02-10T04:03:32.000Z","serviceId":0,"serviceId4k":null,"externalServiceId":322,"externalServiceId4k":null,"externalServiceSlug":"476669","externalServiceSlug4k":null,"ratingKey":"355473","ratingKey4k":null,"plexUrl":"https://app.plex.tv/desktop#!/server/XXXXXXXXXXXXXX","serviceUrl":"http://172.18.100.1:7878/movie/476669"},"modifiedBy":null}]}
```

Here we can see that there are 1 problems currently open by a user : `[...],"results":1,[...]`

**Reply by lissy93** [reply]
Hey @EVOTk 
Yeah, I deffo will implement those widgets, should be fun to build :) Thanks for the useful info, that's really helpful.
I've not come across Overseerr before, but it looks awesome, thanks for sharing.
It might not be for a few weeks, as I've got quite a lot on at the moment, but I will update you with the progress.

**Reply by EVOTk** [reply]
Thank you for your answer, this is great news. I can't wait to see that. I don't know how to develop, but I'm happy to be able to help you!! If you need any other information or test do not hesitate!

**Reply by EVOTk** [reply]
Hello, I just saw this:
https://github.com/Lissy93/dashy/commit/2e63da041ab8a8ddb18a8ef219b4830214654a07

Would you like me to open a [FEATURE_REQUEST] for the Overseerr Widget?

**Reply by CrazyWolf13** [reply]
I'd suggest that, Discussions is currently a bit unsctructured and it's really hard for Lissy to keep on track here.

**Comment by obsidiangroup** [comment]
Has there been any movement on this?  The good thing about radarr and sonarr is, their APIs are very similar so could use a base set of code for both

**Comment by yuriw** [comment]
I’d love to see this implemented too
